pkg/clusteragent/admission/mutate/autoinstrumentation/target_filter.go

```diff
@@ -0,0 +1,115 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package autoinstrumentation
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// TargetFilter filters pods based on a set of targeting rules.
+type TargetFilter struct {
```

Comment:
Why is this a public type? If I'm reading right it has no public methods or fields.

Reply:
Add a TODO for this